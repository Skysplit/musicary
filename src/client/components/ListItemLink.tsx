import React, { ReactNode, PureComponent } from 'react';
import Link from '@app/next/link';
import { LinkProps } from 'next-routes';
import { ListItemProps } from 'react-md';

export type ListItemLinkProps = Partial<LinkProps> & Partial<ListItemProps> & {
  children: ReactNode,
};

export default class ListItemLink extends PureComponent<ListItemLinkProps> {
  render() {
    const { children, route, shallow, prefetch, itemRef, passHref = false, ...props } = this.props;

    return (
      <Link
        passHref={passHref}
        route={route}
        shallow={shallow}
        prefetch={prefetch}
      >
        <div {...props}>
          {children}
        </div>
      </Link>
    );
  }
}
