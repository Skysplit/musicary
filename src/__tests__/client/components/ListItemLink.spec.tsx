
import React from 'react';
import { shallow } from 'enzyme';
import ListItemLink, { ListItemLinkProps } from '@client/components/ListItemLink';

describe('<ListItemLink />', () => {
  test('should render component', () => {
    const props = {
      children: <span>test</span>,
      route: 'test',
      shallow: true,
      prefetch: true,
      passHref: true,
      propsToDiv: 'propInDiv',
    } as ListItemLinkProps;
    const wrapper = shallow(<ListItemLink {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
