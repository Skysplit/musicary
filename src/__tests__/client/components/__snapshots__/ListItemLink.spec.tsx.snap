// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItemLink /> should render component 1`] = `
<LinkRoutes
  passHref={true}
  prefetch={true}
  route="test"
  shallow={true}
>
  <div
    propsToDiv="propInDiv"
  >
    <span>
      test
    </span>
  </div>
</LinkRoutes>
`;
